src/Pidfile.jl

```diff
@@ -29,9 +30,13 @@ Optional keyword arguments:
      The file won't be deleted until 25x longer than this if the pid in the file appears that it may be valid.
      By default this is disabled (`stale_age` = 0), but a typical recommended value would be about 3-5x an
      estimated normal completion time.
+ - `wait`: If true, block until we get the lock, if false, throw PidLockFailedError if lock fails.
 """
 function mkpidlock end
 
+struct PidLockFailedError <: Exception
+    msg::String
+end
```

Comment:
also will need to update the test

```suggestion
 - `wait`: If true, block until we get the lock, if false, throw an exception if lock fails.
"""
function mkpidlock end

```